operator/src/main/java/oracle/kubernetes/operator/helpers/CallBuilder.java

```diff
@@ -1183,6 +1184,29 @@ public Step deletePodAsync(
         responseStep, new RequestParams("deletePod", namespace, name, deleteOptions, domainUid), deletePod);
   }
 
+  /**
```

Comment:
The ability to add a custom retry strategy seems pretty general and you seem to be adding a fair bit of code to make it work only for the deletePod case. A better pattern, I think, would be to add a `withRetryStrategy` method to CallBuilder, analogous to `withFieldSelector` and `withLabelSelectors` and pass it along in that fashion. That way it could be used in other cases. At least look at how difficult that would be to do.